packages/ui-components/src/components/calendar/calendar.tsx

```diff
@@ -0,0 +1,243 @@
+ */
+@Component({
+	tag: 'kv-calendar',
+	styleUrls: {
+		night: 'calendar.night.scss',
+		light: 'calendar.light.scss'
+	},
+	shadow: true
+})
+export class KvCalendar implements ICalendar, ICalendarEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: false }) selectedDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) initialDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) disabledDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) minDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) maxDate?: string;
+
+	/** @inheritdoc */
+	@Event() changeMonth: EventEmitter<IChangeMonthEvent>;
+	/** @inheritdoc */
+	@Event() changeYear: EventEmitter<IChangeYearEvent>;
+	/** @inheritdoc */
+	@Event() clickDate: EventEmitter<IClickDateEvent>;
+
+	@State() month: number = getDateMonth(this.initialDate ?? moment());
+	@State() year: number = getDateYear(this.initialDate ?? moment());
+	@State() hoveredDay: number;
+
+	public onClickPreviousMonth = (event: CustomEvent<MouseEvent>) => {
+		if (this.month === 1) {
+			this.year = this.year - 1;
+			this.month = 12;
+
+			this.changeMonth.emit({ event: event.detail, payload: this.month });
+			this.changeYear.emit({ event: event.detail, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month - 1;
+		this.changeMonth.emit({ event: event.detail, payload: this.month });
+	};
+
+	public onClickNextMonth = (event: CustomEvent<MouseEvent>) => {
+		if (this.month === 12) {
+			this.year = this.year + 1;
+			this.month = 1;
+
+			this.changeMonth.emit({ event: event.detail, payload: this.month });
+			this.changeYear.emit({ event: event.detail, payload: this.year });
+
+			return;
+		}
+
+		this.month = this.month + 1;
+		this.changeMonth.emit({ event: event.detail, payload: this.month });
+	};
+
+	public isDayDisabled = (day: number): boolean => {
+		const dayMoment = fromDateToMoment(day, this.month, this.year);
+
+		if (isDateInArray(dayMoment, this.disabledDates)) {
+			return true;
+		}
+
+		if (this.minDate && isDateBefore(dayMoment, this.minDate)) {
+			return true;
+		}
+
+		if (this.maxDate && isDateAfter(dayMoment, this.maxDate)) {
+			return true;
+		}
+
+		return false;
+	};
+
+	public onClickDay(day: number, event: MouseEvent): void {
+		const clickedDateMoment = fromDateToMoment(day, this.month, this.year);
+		this.clickDate.emit({ event, payload: clickedDateMoment.toISOString() });
+	}
+
+	public isDayActive = (day: number): boolean => {
+		const dayMoment = fromDateToMoment(day, this.month, this.year);
+
+		return isDateInArray(dayMoment, this.selectedDates);
+	};
+
+	public onMouseEnter = (day: number): void => {
+		this.hoveredDay = day;
+	};
+
+	public onMouseLeave = (_day: number): void => {
+		this.hoveredDay = undefined;
+	};
+
+	public getSelectedRange = (): [] | [string] | [string, string] => {
+		if (this.selectedDates && this.selectedDates.length > 0) {
+			const startDate = this.selectedDates[0];
```

Comment:
We can have a range defined like this: `[undefined, undefined]` ?

Reply:
No, the `selectedDates` is an array of strings, for that its values cannot be undefined.